apis/Google.Cloud.Firestore/Google.Cloud.Firestore/FirestoreDb.cs

```diff
@@ -410,8 +420,12 @@ public async Task<T> RunTransactionAsync<T>(Func<Transaction, Task<T>> callback,
                     }
                     catch (RpcException e) when (CheckRetry(e, ref rollback))
```

Comment:
I think there's a bug here that was there before.

If commit fails with aborted and there are attempts left, we never rollback that failed transaction right? Or is it OK not to rollback it because it already failed?

Reply:
I think it's okay, but we'll need to check. We definitely need to make changes here in general though - this PR is only attempting to affect the timing.